webcompat/form.py

```diff
@@ -443,6 +443,28 @@ def add_metadata(form, metadata_dict):
     return form
 
 
+def extract_report_source(form_data):
+    """Extract the source of report.
+
+    This also controls if the src argument is part of a known list.
+    """
+    reported_with = 'unknown'
+    try:
+        reported_with = form_data['src']
+        if not is_valid_source(reported_with):
+            reported_with = 'unknown'
+    except KeyError:
```

Comment:
Just writing this down to make sure I understand it:

if we have a `src` key in the `form_data` dict, and it's not in our `REPORTED_WITH` control list, we set `reported_with` to `unknown`. But if the `src` key isn't even there, we assume it's a `web` report (because that doesn't set one).

LGTM.